spring-integration-core/src/main/java/org/springframework/integration/handler/AbstractMessageHandler.java

```diff
@@ -40,18 +43,31 @@ public abstract class AbstractMessageHandler extends MessageHandlerSupport
 	@Override // NOSONAR
 	public void handleMessage(Message<?> message) {
 		Assert.notNull(message, "Message must not be null");
-		if (isLoggingEnabled() && this.logger.isDebugEnabled()) {
-			this.logger.debug(this + " received message: " + message);
+		if (isLoggingEnabled()) {
+			this.logger.debug(() -> this + " received message: " + message);
 		}
-		MetricsCaptor metricsCaptor = getMetricsCaptor();
-		if (metricsCaptor != null) {
-			handleWithMetrics(message, metricsCaptor);
+		ObservationRegistry observationRegistry = getObservationRegistry();
+		if (observationRegistry != null) {
+			handleWithObservation(message, observationRegistry);
 		}
 		else {
-			doHandleMessage(message);
+			MetricsCaptor metricsCaptor = getMetricsCaptor();
+			if (metricsCaptor != null) {
+				handleWithMetrics(message, metricsCaptor);
+			}
+			else {
+				doHandleMessage(message);
+			}
 		}
 	}
 
+	private void handleWithObservation(Message<?> message, ObservationRegistry observationRegistry) {
+		Observation.createNotStarted(CONSUME_OBSERVATION_NAME, new MessageReceiverContext(message), observationRegistry)
+				.lowCardinalityKeyValue("type", "handler")
```

Comment:
We suggest moving this out to use `DocumentedObservation` together with `ObservationConvention`. That way you give users an option to control the key values, naming etc. + you can get the documentation generated automatically

Reply:
Sounds interesting, but we definitely need to come up with that convention for end-users 😄 .

I think these couple keys I have here so far are mandatory and really specific to component.
I really design an `Observation` usage similar to what we have so far with `Timer` here or there.

The `DocumentedObservation` looks like - will investigate that one.

Thank you!